### PR TITLE
Remove markup-faces from code

### LINKED ISSUE #25: Replace obsolete markup-faces to adoc-faces
Package `markup-faces` is obsolete (latest update 8 years ago). Main idea -- customization font faces for `adoc-mode` like faces for `markdown-mode`: https://github.com/jrblevin/markdown-mode/blob/master/markdown-mode.el#L1827

### PR DESCRIPTION
**Pull request by dunmaksim**
Primary changes:

1. Removed `markup-faces` from requirements.
2. Created `adoc-faces`.
3. Fixed font locks for some text blocks.
4. Fixed some doc strings.
5. Fixed keywords.
6. Added code from `markdown-mode` (for future refactoring).

close #25

### CODE REVIEW

File: adoc-mode.el [hunk 1]
```diff
@@ -552,23 +921,35 @@ To become a customizable variable when regexps for list items become customizabl
...
-1 leading delimiter inclusive whites between delimiter and title text
-2 title's text exclusive leading/trailing whites
-3 trailing delimiter with all whites
-4 trailing delimiter only inclusive whites between title text and delimiter
-0 only chars that belong to the title block element
-
-==  my title  ==  n
----12------23------
-            4--4"
+  When LEVEL is nil, a one line title of any level is matched.
```
Comment (bbatsov) [inline adoc-mode.el:1488]: In Elisp the current indentation level was actually the right one, so please undo this type of changes.

**Comment by bbatsov** [comment]
> Added code from markdown-mode (for future refactoring).

I don't quite understand this part. 

I've also noticed a few unrelated changes in both commits - e.g. in the first commit there are some changes to docstring indentation that look incorrect to me. In the second there are also changes that are unrelated to the markup-faces. My suggestion would be to limit this PR to the removal of markup-faces and to submit other suggestions for improvements separately, so that the review process would be simplified and the changes would stay focused.

File: adoc-mode.el (hunk 1 above)
Comment (dunmaksim) [reply adoc-mode.el]: Hello! Fixed.

**Comment by dunmaksim** [comment]
> > Added code from markdown-mode (for future refactoring).
> 
> I don't quite understand this part.
> 
> I've also noticed a few unrelated changes in both commits - e.g. in the first commit there are some changes to docstring indentation that look incorrect to me. In the second there are also changes that are unrelated to the markup-faces. My suggestion would be to limit this PR to the removal of markup-faces and to submit other suggestions for improvements separately, so that the review process would be simplified and the changes would stay focused.

Hello!

Diff cleaned from non-remove-markup edits.

**Comment by dunmaksim** [comment]
@bbatsov , hello! Please, do review of updated PR.

File: adoc-mode.el [hunk 2]
```diff
@@ -8,8 +8,8 @@
 ;; Maintainer: Bozhidar Batsov <bozhidar@batsov.dev>
 ;; Created: 2009
 ;; Version: 0.7.0-snapshot
-;; Package-Requires: ((emacs "26") (markup-faces "1.0.0"))
-;; Keywords: AsciiDoc
+;; Package-Requires: ((emacs "26"))
+;; Keywords: docs, wp
```
Comment (bbatsov) [inline adoc-mode.el:12]: This change here seems like a mistake to me.

File: adoc-mode.el [hunk 3]
```diff
@@ -34,20 +34,19 @@
 ;; writing short documents, articles, books and UNIX man pages.  AsciiDoc files
 ;; can be translated to HTML and DocBook markups.
 ;;
-;; adoc-mode is an Emacs major mode for editing AsciiDoc files.  It emphasizes on
-;; the idea that the document is highlighted so it pretty much looks like the
+;; `adoc-mode' is an Emacs major mode for editing AsciiDoc files.  It emphasizes on
+;; the idea that the document is highlighted so it pretty much ;; looks like the
```
Comment (bbatsov) [inline adoc-mode.el:38]: Same here - two lines got joined by mistake.

File: adoc-mode.el (hunk 2 above)
Comment (dunmaksim) [reply adoc-mode.el:12]: My static analyzer write:

**You should include standard keywords: see the variable `finder-known-keywords'.**

I follow into the file `finder.el` and see variable `finder-known-keywords` (from my EMACS 27.1):

```elisp
;; These are supposed to correspond to top-level customization groups,
;; says rms.
(defvar finder-known-keywords
  '((abbrev	. "abbreviation handling, typing shortcuts, and macros")
    (bib	. "bibliography processors")
    (c		. "C and related programming languages")
    (calendar	. "calendar and time management tools")
    (comm	. "communications, networking, and remote file access")
    (convenience . "convenience features for faster editing")
    (data	. "editing data (non-text) files")
    (docs	. "Emacs documentation facilities")
    (emulations	. "emulations of other editors")
    (extensions	. "Emacs Lisp language extensions")
    (faces	. "fonts and colors for text")
    (files      . "file editing and manipulation")
    (frames     . "Emacs frames and window systems")
    (games	. "games, jokes and amusements")
    (hardware	. "interfacing with system hardware")
    (help	. "Emacs help systems")
    (hypermedia . "links between text or other media types")
    (i18n	. "internationalization and character-set support")
    (internal	. "code for Emacs internals, build process, defaults")
    (languages	. "specialized modes for editing programming languages")
    (lisp	. "Lisp support, including Emacs Lisp")
    (local	. "code local to your site")
    (maint	. "Emacs development tools and aids")
    (mail	. "email reading and posting")
    (matching	. "searching, matching, and sorting")
    (mouse	. "mouse support")
    (multimedia . "images and sound")
    (news	. "USENET news reading and posting")
    (outlines   . "hierarchical outlining and note taking")
    (processes	. "processes, subshells, and compilation")
    (terminals	. "text terminals (ttys)")
    (tex	. "the TeX document formatter")
    (tools	. "programming tools")
    (unix	. "UNIX feature interfaces and emulators")
    (vc		. "version control")
    (wp		. "word processing"))
  "Association list of the standard \"Keywords:\" headers.
Each element has the form (KEYWORD . DESCRIPTION).")
```

File: adoc-mode.el (hunk 3 above)
Comment (dunmaksim) [reply adoc-mode.el]: Fixed.

File: adoc-mode.el (hunk 2 above)
Comment (bbatsov) [reply adoc-mode.el:12]: Hmm, that's something new to me, but I guess it makes some sense.

File: adoc-mode.el [hunk 4]
```diff
@@ -123,6 +124,38 @@ need to call `adoc-calc' for the change to take effect."
                  function)
   :group 'adoc)
 
+(defcustom adoc-hide-markup nil
```
Comment (bbatsov) [inline adoc-mode.el:127]: I think you copied this one from Markdown mode, but it's never used anywhere in your PR.

File: adoc-mode.el [hunk 5]
```diff
@@ -123,6 +124,38 @@ need to call `adoc-calc' for the change to take effect."
...
+Functions such as `htmlfontify-buffer' and `ps-print-buffer' will
+not honor these text properties.  For printing, it would be better
+to first convert to HTML or PDF (e.g,. using Pandoc)."
+  :group 'adoc
+  :type 'boolean
+  :safe 'booleanp
+  :package-version '(adoc-mode . "0.7.0"))
+(make-variable-buffer-local 'adoc-hide-markup)
+
+(defcustom adoc-uri-types
```
Comment (bbatsov) [inline adoc-mode.el:149]: Can't see other references to this new defcustom either.

File: adoc-mode.el [hunk 6]
```diff
@@ -56,36 +55,38 @@
...
-there.  adoc-mode has only a few faces of its own, which can be
-customized on this page."
+(defgroup adoc nil "Support for editing AsciiDoc files in GNU Emacs."
   :group 'text
   :prefix "adoc-"
+  :version "0.7.0"
   :link '(url-link "https://github.com/bbatsov/adoc-mode"))
 
+
+(defcustom adoc-mode-hook nil
```
Comment (bbatsov) [inline adoc-mode.el:66]: I think this hook automatically defined by the major mode, so it's not really needed as an explicit variable.

File: adoc-mode.el [hunk 7]
```diff
@@ -540,8 +570,452 @@ To become a customizable variable when regexps for list items become customizabl
...
+(defvar adoc-subscript-face 'adoc-subscript-face)
+
+(defface adoc-highlighting-face
+  '((t (:background "yellow" :foreground "black")))
+  "Face for highlighting."
+  :group 'adoc-faces)
+(defvar adoc-highlighting-face 'adoc-highlighting-face)
+
+(defface adoc-title-face
+  '((t (:inherit adoc-markdown-face :weight bold)))
```
Comment (bbatsov) [inline adoc-mode.el:955]: What is this `adoc-markdown-face`?

File: adoc-mode.el [hunk 8]
```diff
@@ -540,8 +570,452 @@ To become a customizable variable when regexps for list items become customizabl
...
+  :group 'adoc-faces)
+(defvar adoc-pre-face 'adoc-pre-face)
+
+(defface adoc-table-face
+  '((t (:inherit (adoc-code-face))))
+  "Face for tables."
+  :group 'adoc-faces)
+(defvar adoc-table-face 'adoc-table-face)
+
+(defface adoc-language-keyword-face
```
Comment (bbatsov) [inline adoc-mode.el:802]: Hmm, I don't see any usages of this face and I assume of some other faces as well. Were those copied from markdown-mode?

I think it'd be best to limit this PR just to things that are actually used right now and do any additions/improvements subsequently in other PRs.

**Comment by bbatsov** [comment]
I took a closer look at the PR today and while it looks better it seems it still features some code (defcustoms, faces) that was copied from markdown-mode, but is not currently used. Please, limit the scope of the changes only to things that are used, as I'm not fond of adding unused code.

File: adoc-mode.el (hunk 4 above)
Comment (dunmaksim) [reply adoc-mode.el]: Removed.

File: adoc-mode.el (hunk 5 above)
Comment (dunmaksim) [reply adoc-mode.el]: Removed.

File: adoc-mode.el (hunk 6 above)
Comment (dunmaksim) [reply adoc-mode.el]: Removed.

File: adoc-mode.el (hunk 7 above)
Comment (dunmaksim) [reply adoc-mode.el]: Removed.

File: adoc-mode.el (hunk 8 above)
Comment (dunmaksim) [reply adoc-mode.el:720]: All unused faces removed. Some existing faces now inherits from faces from `font-lock.el`.

**Comment by dunmaksim** [comment]
@bbatsov , all unused code from `markdown-mode` removed. Some existing faces now inherit from faces from `font-lock.el` module, for example:

```elisp
((t :inherit adoc-gen-face :weight bold)) --> ((t :inherit (adoc-gen-face bold))
```

Where `bold` is face from `font-lock.el`.

File: adoc-mode.el [hunk 9]
```diff
@@ -133,7 +126,8 @@ configuration file."
           (string :tag "level 1")
           (string :tag "level 2")
           (string :tag "level 3")
-          (string :tag "level 4") )
+          (string :tag "level 4")
+          (string :tag "level 5"))
```
Comment (dunmaksim) [inline adoc-mode.el:130]: Added level 5 from [AsciiDoc documentation](https://docs.asciidoctor.org/asciidoc/latest/sections/titles-and-levels/).

File: adoc-mode.el [hunk 10]
```diff
@@ -352,22 +316,34 @@ To become a customizable variable when regexps for list items become customizabl
...

+
 ;;;; help text copied from asciidoc manual
 (defconst adoc-help-constrained-quotes
   "Constrained quotes must be bounded by white space or commonly
   adjoining punctuation characters. These are the most commonly
   used type of quote.")
 (defconst adoc-help-emphasis
   "Usually rendered italic")
-(defconst adoc-help-strong
```
Comment (dunmaksim) [inline adoc-mode.el:363]: Now AsciiDoc use word **bold** instead **strong**: https://docs.asciidoctor.org/asciidoc/latest/text/bold/

**Comment by bbatsov** [comment]
Thanks for working on this!

**Pull request merged**